### PR TITLE
Experiment tracking with W&B

### PR DESCRIPTION
**Pull request by dmitryduev**
This PR adds basic experiment tracking with [Weights & Biases](https://wandb.com) for DNN training.

image

### CODE REVIEW

File: scope/utils.py [hunk 1]
```diff
@@ -350,7 +350,7 @@ def plot_gaia_density(
     # ecliptic
     for ecllat in zip([0, -30, 30], [2, 1, 1]):
         theta = np.arange(0.0, 360, 0.036)
-        phi = gallat * np.ones_like(theta)
+        phi = ecllat * np.ones_like(theta)
```
Comment (dmitryduev) [inline scope/utils.py:353]: @mcoughlin was that a bug?

Comment (mcoughlin) [reply scope/utils.py:353]: Who me make bugs? ;) I guess it didnt come up as it is just a question of phi's, but good to fix.

File: config.defaults.yaml [hunk 2]
```diff
@@ -961,24 +964,16 @@ feature_stats:
...
-      # label name from the dataset csv file
+      # value of label should refer to dataset.dN.csv
       label: variable
+      # conv_branch: false
       # value should refer to features section of this config
       features: phenomenological
       # our "labels" are floats [0., 0.25, 0.5, 0.75, 1.]
-      threshold: 0.7
-      # balance ratio for the prevalent class. leave as null to use all available data
+      threshold: 0.9
```
Comment (mcoughlin) [inline config.defaults.yaml:975]: Why was this changed?

File: doc/usage.md [hunk 3]
```diff
@@ -33,3 +33,21 @@ but could also be overridden with optional `scope.py train` arguments, e.g.
 
 - Feature stats to be used for feature scaling/standardization before training
   is defined in `config.yaml` under `feature_stats`.
+
```
Comment (mcoughlin) [inline doc/usage.md:36]: This is really nice.

File: scope.py [hunk 4]
```diff
@@ -603,6 +605,31 @@ def train(
         if pre_trained_model is not None:
             classifier.load(pre_trained_model)
 
+        time_tag = datetime.datetime.utcnow().strftime("%Y%m%d_%H%M%S")
+
+        if not kwargs.get("test", False):
+            wandb.login(key=self.config["wandb"]["token"])
```
Comment (mcoughlin) [inline scope.py:611]: A link to where to documentation for where to grab one's token?

**Review by mcoughlin — APPROVED** [review]
Looks like. Might want a little more pointer to w+b doc's stuff.

File: scope.py (hunk 4 above)
Comment (dmitryduev) [reply scope.py:611]: W&B's wizard will actually print out the url if you're running it the first time from the command line. Added anyways.

**Comment by dmitryduev** [comment]
@mcoughlin mind merging once the CI is green?

**Comment by dannygoldstein** [comment]
😍

**Comment by mcoughlin** [comment]
@dmitryduev I merge now?

**Comment by dmitryduev** [comment]
@mcoughlin actually, let me add one more thing (I don't track test acc/precision/etc). I'll ping you.

**Comment by dmitryduev** [comment]
@mcoughlin should be ready for re-review.

**Comment by mcoughlin** [comment]
Beautiful.
